analysisCode/src/TruthEvent.cpp

```diff
@@ -1,8 +1,82 @@
 #include "include/TruthEvent.h"
-
 #include <iostream>
 
-TruthEvent::TruthEvent()
+TruthEvent::TruthEvent(){cout << "A truth event" << endl;}
+
+void TruthEvent::ProcessEvent(erhic::EventPythia* ev, JetDef def, SoftDropJetDef softdrop_def)
+{ 
+  contrib::SoftDrop softdrop = softdrop_def.getSoftDrop(); 
+  Set_truthEvent(ev);
+  //PrintEvent();
+  ParticleLoop(ev);
+  JetLoop(def, softdrop);   
+ }
+
+void TruthEvent::ParticleLoop(erhic::EventPythia* ev) 
 {
-  std::cout<<"A truth event"<<std::endl;
+  // initialize variables for boost to Breit frame -- this will be moved
+  double Px,Py,Pz,P0,qx,qy,qz,q0;
+  for (int part = 0; part < nTracks; part++)
+    {
+      const Particle *truthPart = ev->GetTrack(part);
+      Set_truthPart(truthPart);
+
+      // Block for boost to Breit frame -- this will be moved //
+      if (part==1)
+	{
+	  Px = part_px; Py = part_py; Pz = part_pz; P0 = part_E;
+	}
+      if (part==3) 
+	{ 
+	  qx = part_px; qy = part_py; qz = part_pz; q0 = part_E;
+	}
+      double E_breit = 2*true_x*P0 + q0;
+      TVector3 breit((2*true_x*Px+qx)/E_breit,(2*true_x*Py+qy)/E_breit, (2*true_x*Pz+qz)/E_breit);
+      //PrintParticle();
+      TLorentzVector part_4vec = truthPart->PxPyPzE();
+      //cout << " Lab frame... " << endl;
+      //part_4vec.Print();
+      part_4vec.Boost(breit);
+      //cout << " Breit frame... " << endl; 
+      //part_4vec.Print();
+      truthPseudoJets.push_back(PseudoJet(part_px, part_py, part_pz, part_E));
+      //truthPseudoJets.push_back(PseudoJet(part_4vec));
+
+      // End Breit Block //
+    }
+  
 }
+
+void TruthEvent::JetLoop(JetDef def, contrib::SoftDrop sd) 
+{ 
+  Selector selectRapidity = SelectorAbsRapMax(4);
```

Comment:
These should be taken from the `JetDef` class - in fact perhaps we should just put a `Selector` member variable into `JetDef` that has an accessor which contains this information.

Reply:
Yes I was planning on coming back to this -- I hadn't yet figured out how to make cuts with the JetDef class you made, but it should be easy enough for me to put a Selector member variable into JetDef as you suggested.